packages/cli/src/commands/generate/page/__tests__/page.test.js

```diff
@@ -106,3 +162,97 @@ test('creates a path equal to passed path', () => {
     '<Route path="fooBar-baz" page={FooBarPage} name="fooBar" />',
   ])
 })
+
+test('file generation', async () => {
+  mockFiles = {
+    [getPaths().web.routes]: [
+      '<Router>',
+      '  <Route path="/about" page={AboutPage} name="about" />',
+      '  <Route notfound page={NotFoundPage} />',
+      '</Router>',
+    ].join('\n'),
+  }
+
+  const spy = jest.spyOn(fs, 'writeFileSync')
+  mockFs = true
+
+  await page.handler({ name: 'home', path: '', force: false })
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/pages/HomePage/HomePage.js'),
+    loadGeneratorFixture('page', 'singleWordPage.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/pages/HomePage/HomePage.test.js'),
+    loadGeneratorFixture('page', 'singleWordPage.test.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize(
+      '/path/to/project/web/src/pages/HomePage/HomePage.stories.js'
+    ),
+    loadGeneratorFixture('page', 'singleWordPage.stories.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/Routes.js'),
+    [
+      '<Router>',
+      '  <Route path="/home" page={HomePage} name="home" />',
+      '  <Route path="/about" page={AboutPage} name="about" />',
+      '  <Route notfound page={NotFoundPage} />',
+      '</Router>',
+    ].join('\n')
+  )
+
+  mockFs = false
+  spy.mockRestore()
+})
+
+test('file generation with route params', async () => {
+  mockFiles = {
+    [getPaths().web.routes]: [
+      '<Router>',
+      '  <Route path="/about" page={AboutPage} name="about" />',
+      '  <Route notfound page={NotFoundPage} />',
+      '</Router>',
+    ].join('\n'),
+  }
+
+  const spy = jest.spyOn(fs, 'writeFileSync')
+  mockFs = true
+
+  await page.handler({ name: 'post', path: '{id}', force: false })
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/pages/PostPage/PostPage.js'),
+    loadGeneratorFixture('page', 'paramPage.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/pages/PostPage/PostPage.test.js'),
+    loadGeneratorFixture('page', 'paramPage.test.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize(
+      '/path/to/project/web/src/pages/PostPage/PostPage.stories.js'
+    ),
+    loadGeneratorFixture('page', 'paramPage.stories.js')
+  )
+
+  expect(spy).toHaveBeenCalledWith(
+    path.normalize('/path/to/project/web/src/Routes.js'),
+    [
+      '<Router>',
```

Comment:
This isn't close enough to what the actual Routes.js file would look like. (Also see comment further down, with the regex stuff).

Reply:
Fixed